src/js/profile/mobile/widget/Calendar.js

```diff
@@ -0,0 +1,275 @@
+ * @extends ns.widget.BaseWidget
+ * @author Dohyung Lim <delight.lim@samsung.com>
+ */
+(function (document, ns) {
+	"use strict";
+	//>>excludeStart("tauBuildExclude", pragmas.tauBuildExclude);
+	define(
+		[
+			"../../../core/engine",
+			"../../../core/util/DOM/attributes",
+			"../widget"
+		],
+		function () {
+			//>>excludeEnd("tauBuildExclude");
+			var utilsObject = ns.util.object,
+				engine = ns.engine,
+				events = ns.event,
+				weeks = ["MON", "TUE", "WED", "THU", "FRI", "SAT", "SUN"],
+				fullNameMonth = ["January", "February", "March", "April", "May", "June", "July", "August", "September", "October", "November", "December"],
+				today = new Date(),
+				todayYear = today.getFullYear(),
+				todayMonth = today.getMonth() + 1,
+
+				Calendar = function () {
+					this.options = utilsObject.merge({}, Calendar.defaults);
+				},
+
+				defaults = {
+					value: false // false is past selectable, true is past selection not possible.
+				},
+
+				BaseWidget = ns.widget.BaseWidget,
+				prototype = new BaseWidget();
+
+			Calendar.prototype = prototype;
+			Calendar.defaults = defaults;
+
+			prototype._init = function (element) {
+				var self = this;
+
+				if (!element.getAttribute("value")) {
+					element.setAttribute("value", self.options.value);
+				}
+
+				prototype._buildCalendar();
+
+				return element;
+			};
+
+			prototype._buildCalendar = function () {
+				var firstDate = new Date(todayYear, todayMonth - 1, 0);
+				var lastDate = new Date(todayYear, todayMonth, 0);
+				var day = firstDate.getDay();
+				var prevLastDate = lastDate.getDate();
+				var week = Math.ceil(prevLastDate / 7) + 1;
+				var leftDays = 7;
+				var setDays = 1;
+				var nextMonthDate = 1;
+				var cell = null;
+				var calendar = document.getElementById("calendar");
+				var marginRow = calendar.insertRow();
+
+				marginRow.style.height = "7px";
+
+				for (var i = 1; i < week + 1; i++) {
+					var row = calendar.insertRow();
+
+					while (day != 0) { // prev disabled days.
+						day = day - 1;
+						leftDays = leftDays - 1;
+						cell = row.insertCell();
+						cell.innerHTML = prevLastDate - day;
+						cell.style.opacity = "10%";
+					}
+					while (leftDays != 0) {
+						if (setDays > prevLastDate) { // next disabled days.
+							cell = row.insertCell();
+							cell.innerHTML = nextMonthDate;
+							cell.style.opacity = "40%";
+							leftDays = leftDays - 1;
+							nextMonthDate = nextMonthDate + 1;
+						} else { // current enabled days.
+							cell = row.insertCell();
+							cell.innerHTML = setDays;
+							cell.style.opacity = "100%";
+							setDays = setDays + 1;
+							leftDays = leftDays - 1;
+						}
+					}
+					leftDays = 7;
+				}
+
+				document.getElementById("yearMonth").innerHTML = fullNameMonth[todayMonth - 1] + " " + todayYear;
+			}
+
+			prototype._deleteCalendar = function () {
+				var calendar = document.getElementById("calendar");
+
+				while (calendar.rows.length > 2) {
```

Comment:
Is there no return condition?
Is the possible that the length is always bigger than '2'?

Reply:
By default, row has the 'top row' and 'the day of the week row'.
Therefore, the two are not always fixed.
The conclusion cannot be less than 2.